ethash.go

```diff
@@ -104,14 +105,25 @@ func freeCache(cache *cache) {
 	cache.ptr = nil
 }
 
-// Light implements the Verify half of the proof of work.
-// It uses a small in-memory cache to verify the nonces
-// found by Full.
+// Light implements the Verify half of the proof of work. It uses a few small
+// in-memory caches to verify the nonces found by Full.
 type Light struct {
-	test    bool       // if set use a smaller cache size
-	mu      sync.Mutex // protects current
-	current *cache     // last cache which was generated.
-	// TODO: keep multiple caches.
+	test  bool              // if set use a smaller cache size
+	mu    sync.Mutex        // protects the per-epoch map of DAGs
+	dags  map[uint64]*cache // currently cached verification DAGs
```

Comment:
Can we rename this to "caches"? Ethash terminology distinguishes cache (for verification) and DAG (for mining).